tesseract_kinematics/src/core/inverse_kinematics.cpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * @file inverse_kinematics.h
+ * @brief Inverse kinematics functions.
+ *
+ * @author Levi Armstrong
+ * @date April 15, 2018
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2013, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <tesseract_kinematics/core/utils.h>
+#include <tesseract_kinematics/core/inverse_kinematics.h>
+
+namespace tesseract_kinematics
+{
+IKSolutions InverseKinematics::getRedundantSolutions(const Eigen::Ref<const Eigen::VectorXd>& sol) const
+{
+  return tesseract_kinematics::getRedundantSolutions<double>(
+      sol, getLimits().joint_limits, getRedundancyCapableJointIndices());
```

Comment:
Seems a bit odd to have such a simple wrapper around a helper function, especially if `getRedundancyCapableJointIndices()` is a public member of this interface. Is the benefit of adding this to the interface just convenience?

Reply:
I added it just for convenience.

Reply:
I was on the fence about adding this. It is not used very many places so I would be ok with removing it. Having here though provided more visibility than as a separate utility function.

Reply:
I'm personally opposed to code duplication in the interest of visibility. If the helper function we have right now is not easy to find we should move it to a better place or document it better. I would suggest putting it as a static member of the IK class, but the FK class also needs to use it. I think that's why I put it in the utils file to begin with

Reply:
I am good with that. I will remove it and I will add documentation to the `calcInvKin` methods.